test/compatibility/suite/test_flcs_deprecate.py

```diff
@@ -0,0 +1,79 @@
+#
+# This is free and unencumbered software released into the public domain.
+#
+# Anyone is free to copy, modify, publish, use, compile, sell, or
+# distribute this software, either in source code form or as a compiled
+# binary, for any purpose, commercial or non-commercial, and by any
+# means.
+#
+# In jurisdictions that recognize copyright laws, the author or authors
+# of this software dedicate any and all copyright interest in the
+# software to the public domain. We make this dedication for the benefit
+# of the public at large and to the detriment of our heirs and
+# successors. We intend this dedication to be an overt act of
+# relinquishment in perpetuity of all present and future rights to this
+# software under copyright law.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+# MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.
+# IN NO EVENT SHALL THE AUTHORS BE LIABLE FOR ANY CLAIM, DAMAGES OR
+# OTHER LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE,
+# ARISING FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+import compatibility_test, errno, wiredtiger
+
+
+class test_flcs_deprecate(compatibility_test.CompatibilityTestCase):
+    '''
+    Test FLCS deprecation handling during database upgrade.
+    '''
+
+    conn_config = ''
+    create_config = 'key_format=r,value_format=8t'
+    uri = 'table:test_flcs_deprecate'
+    nrows = 100
+
+    def test_flcs_deprecate(self):
+        # Only run this test for newer branch where FLCS is deprecated.
+        if self.newer_branch == "feature-flcs":
+            # Run the older-branch part (create the FLCS table and populate it)
+            self.run_method_on_branch(self.older_branch, 'on_older_branch')
+            # Run the newer-branch part (attempt to open and expect failure)
+            self.run_method_on_branch(self.newer_branch, 'on_newer_branch')
+
+    def on_older_branch(self):
+        # This runs in the older branch to create a FLCS table and populate it.
+        conn = wiredtiger.wiredtiger_open('.', 'create,' + self.conn_config)
+        session = conn.open_session()
+
+        session.create(self.uri, self.create_config)
+
+        c = session.open_cursor(self.uri)
+        for i in range(1, self.nrows + 1):
+            c[i] = i
+        c.close()
+        session.close()
+        conn.close()
+
+    def on_newer_branch(self):
+        # Expect opening the FLCS table to fail because FLCS is deprecated.
+        conn = wiredtiger.wiredtiger_open('.', self.conn_config)
+        session = conn.open_session()
+
+        print("Attempting to open cursor on FLCS table, expecting failure...")
+
+        # Expect an exception when trying to open a cursor on the FLCS table.
+        try:
+            session.open_cursor(self.uri)
+        except wiredtiger.WiredTigerError as e:
```

Comment:
Should we ensure we never go beyond the `open_cursor` call?
```suggestion
            session.open_cursor(self.uri)
            assert False
        except wiredtiger.WiredTigerError as e:
```

Reply:
I've added assert as part of [96fa099](https://github.com/wiredtiger/wiredtiger/pull/12818/commits/96fa099591b81a26c66a77257c450ba886ec180e)